addons/mail/sfu_server/src/services/http.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * @type {Map<string, {file: Buffer | undefined, md5: string | undefined}>}
+ */
+const fileCache = new Map();
+
+function cacheFile(filename) {
+    const path = process.cwd() + "/dist/" + filename;
+    const file = fs.readFileSync(path);
+    const md5 = crypto.createHash("md5").update(file).digest("hex");
+    logger.verbose(`${filename} cached with md5: ${md5}`);
+    fileCache.set(filename, { file, md5 });
+}
+
+/**
+ * @param {Object} [options]
+ * @param {string} [options.ip]
+ * @param {number} [options.port]
+ */
+export async function start({ ip, port } = {}) {
+    const listenIp = ip ?? config.IP;
+    const listenPort = port ?? config.PORT;
+    logger.info("starting...");
+    const routeListener = new RouteListener();
+    routeListener.get("/src/<fileName>", {
+        cors: "*",
+        callback: (req, res, { fileName }) => {
+            if (!fileCache.has(fileName)) {
+                return res.end();
+            }
+            const cached = fileCache.get(fileName);
+            res.setHeader("Content-Type", "application/javascript");
+            res.setHeader("Cache-Control", "max-age=60");
+            res.setHeader("ETag", cached.md5);
+            if (req.headers["if-none-match"] === cached.md5) {
+                res.statusCode = 304; // Not modified
+                return res.end();
+            }
+            res.statusCode = 200;
+            return res.end(cached.file);
+        },
+    });
+    routeListener.get("/channel", {
+        cors: "*",
+        callback: async (req, res) => {
+            const remoteAddress = req.headers["x-forwarded-for"] || req.connection.remoteAddress;
+            try {
+                const jsonWebToken = req.headers.authorization?.split(" ")[1];
+                /** @type {{ iss: string }} */
+                const claims = await auth.verify(jsonWebToken);
+                if (!claims.iss) {
+                    logger.warn(`${remoteAddress}: missing issuer claim when creating channel`);
+                    return res.end();
+                }
+                const channel = await Channel.create();
+                logger.info(`channel ${channel.uuid} created for ${claims.iss}(${remoteAddress})`);
+                res.setHeader("Content-Type", "application/json");
+                res.statusCode = 200;
+                return res.end(JSON.stringify(channel.uuid));
+            } catch (error) {
+                logger.warn(`[${remoteAddress}] failed to create channel: ${error.message}`);
+            }
+            return res.end();
+        },
+    });
+    routeListener.post("/disconnect", {
+        cors: config.API_CORS,
+        callback: async (req, res) => {
+            const remoteAddress = req.socket?.remoteAddress;
+            try {
+                const { jsonWebToken } = await parseBody(req, { json: true });
+                /** @type {{ sessionIdsByChannel: Object<string, number[]> }} */
+                const claims = await auth.verify(jsonWebToken);
```

Comment:
So if I understand correctly, `jsonWebToken` contains the parameters + the signature, hence allowing the verification of the whole body.
(which would explain why the signature of GET and POST are handled differently)